velox/docs/functions/spark/regexp.rst

```diff
@@ -38,7 +38,12 @@ See https://github.com/google/re2/wiki/Syntax for more information.
 .. spark:function:: regexp_replace(string, pattern, overwrite) -> varchar
 
     Replaces all substrings in ``string`` that match the regular expression ``pattern`` with the string ``overwrite``. If no match is found, the original string is returned as is.
-    There is a limit to the number of unique regexes to be compiled per function call, which is 20.
+    There is a limit to the number of unique regexes to be compiled per function call, which is 20. If this limit is exceeded the function will throw an exception.
```

Comment:
This documentation applies to all regex functions. Instead repeating it for each function, perhaps, move it to the top of this document.

Reply:
As of now, regexp_extract and rlike call the "ensureRegexIsContstantAndCompatible" function. In that case, would they not be limited to a single regex? I may be unaware, but I am not sure of a use case where they would encounter this exception. I wrote up a quick test for this, results displayed below.
<img width="354" alt="image" src="https://github.com/facebookincubator/velox/assets/14911410/c34f5201-c69a-473d-8d58-0063f35d9aa2">
<img width="278" alt="image" src="https://github.com/facebookincubator/velox/assets/14911410/b58adcdf-e0e6-41aa-9af8-4e63a32ef833">
<img width="350" alt="image" src="https://github.com/facebookincubator/velox/assets/14911410/1146e9e6-4dcd-4594-940b-900d5ac4da6e">

Reply:
> would they not be limited to a single regex?

I see. Some functions support only constant regex and other support variable regex (up to 20 unique). Let's make sure this is clear in the docs.